core/transaction/validator.go

```diff
@@ -129,6 +131,31 @@ func VerifyTransactionWithBlock(iterator Iterator) ErrCode {
 					return ErrSummaryAsset
 				}
 			}
+		case RegisterName:
+			namePayload := txn.Payload.(*payload.RegisterName)
+
+			name := namePayload.Name
+			if _, ok := registeredNames[name]; ok {
+				log.Warn("[VerifyTransactionWithBlock], duplicate name exist in block.")
+				return ErrDuplicateName
+			}
+			registeredNames[name] = struct{}{}
+
+			registrant := BytesToHexString(namePayload.Registrant)
+			if _, ok := nameRegistrants[registrant]; ok {
+				log.Warn("[VerifyTransactionWithBlock], duplicate registrant exist in block.")
+				return ErrDuplicateName
+			}
+			nameRegistrants[registrant] = struct{}{}
+		case DeleteName:
+			namePayload := txn.Payload.(*payload.RegisterName)
+
+			registrant := BytesToHexString(namePayload.Registrant)
+			if _, ok := nameRegistrants[registrant]; ok {
+				log.Warn("[VerifyTransactionWithBlock], duplicate registrant exist in block.")
+				return ErrDuplicateName
+			}
+			nameRegistrants[registrant] = struct{}{}
```

Comment:
Does this mean we won't allow someone to delete and register a name in the same block?

Reply:
yes, because there is no guarantee that they will be persisted in the correct order

Reply:
OK sounds reasonable